app/app.py

```diff
@@ -0,0 +1,190 @@
+import os
+import pickle
+import tempfile
+import uuid
+import dash_bootstrap_components as dbc
+import dash_uploader as du
+from dash import Dash
+from dash import Input
+from dash import Output
+from dash import clientside_callback
+from dash import dcc
+from dash import html
+
+
+dbc_css = "https://cdn.jsdelivr.net/gh/AnnMarieW/dash-bootstrap-templates/dbc.min.css"
+app = Dash(__name__, external_stylesheets=[dbc.themes.UNITED, dbc_css, dbc.icons.FONT_AWESOME])
+
+# ------------------ Nav Bar ------------------ #
+color_mode_switch = html.Span(
+    [
+        dbc.Label(className="fa fa-moon", html_for="color-mode-switch"),
+        dbc.Switch(
+            id="color-mode-switch",
+            value=False,
+            className="d-inline-block ms-1",
+            persistence=True,
+        ),
+        dbc.Label(className="fa fa-sun", html_for="color-mode-switch"),
+    ],
+    className="p-2",
+)
+
+# Define the navigation bar
+navbar = dbc.Row(
+    dbc.Col(
+        dbc.NavbarSimple(
+            children=[
+                dbc.NavItem(dbc.NavLink("Doc", href="https://nplinker.github.io/nplinker/latest/")),
+                dbc.NavItem(
+                    dbc.NavLink("About", href="https://github.com/NPLinker/nplinker-webapp"),
+                ),
+                dbc.NavItem(
+                    color_mode_switch,
+                    className="mt-1 p-1",
+                ),
+            ],
+            brand="NPLinker Webapp",
+            brand_href="https://github.com/NPLinker/nplinker-webapp",
+            color="primary",
+            className="p-3 mb-2",
+            dark=True,
+        ),
+    ),
+)
+
+# ------------------ Uploader ------------------ #
+# Configure the upload folder
+TEMP_DIR = tempfile.mkdtemp()
+du.configure_upload(app, TEMP_DIR)
+
+uploader = html.Div(
+    [
+        dbc.Row(
+            dbc.Col(
+                du.Upload(
+                    id="dash-uploader",
+                    text="Import Data",
+                    text_completed="Uploaded: ",
+                    filetypes=["pkl"],
```

Comment:
```suggestion
                    filetypes=["pkl", "pickle"],
```
Both are common extensions for python pickle file. And it seems `.pickle` is preferred in python 3 doc.